config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -101,6 +107,19 @@ public Value getConfig(ConfigResource configResource) throws IOException {
     return config;
   }
 
+  @Override
+  public List<ContextSpecificConfig> getAllConfigs(String resourceName, String resourceNamespace,
+      String tenantId) throws IOException {
+    List<ContextSpecificConfig> contextSpecificConfigList = new ArrayList<>();
+    for (String context : getAllContexts(resourceName, resourceNamespace, tenantId)) {
```

Comment:
Can we optimize this? We're doing N+1 separate DB queries here if I'm reading this right, getAll should be doable as a single query. Edit: is this what the todo below is referring to?

Reply:
Updated it.
With the previous approach, I was trying to minimize the no. of docs processed but had to make N+1 queries, as you pointed out. But that would have been possible with the distinct clause only. Without distinct clause, we'd still have to process more documents(the way we're doing it now).
With the updated approach, we're making 1 query only but have to process all versions of a config resource even though we want the latest one.